### PR TITLE
Update GCC to 13.2, Clang to 16.0.0, Bazel to 7.0.0

### PR DESCRIPTION
**Pull request by Sckorn**
Update compiler and build system versions in docker.

### CODE REVIEW

File: docker/dockerfiles/build_clang.docker [hunk 1]
```diff
@@ -6,42 +6,46 @@ FROM radameon/build_${BASE}:latest
 
 RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 
-# Installing all clang 14 packages
+# Installing all clang 17 packages
```
Comment (radameon) [inline docker/dockerfiles/build_clang.docker:9]: Clang 16, you need to fix this.

**Review by radameon — CHANGES_REQUESTED** [review]
Minor change reuired, and we're good

File: docker/dockerfiles/build_clang.docker (hunk 1 above)
Comment (Sckorn) [reply docker/dockerfiles/build_clang.docker]: Fixed

**Review by radameon — APPROVED** [review]
LGTM

**Pull request merged**